orchagent/routeorch.cpp

```diff
@@ -753,7 +772,7 @@ void RouteOrch::doTask(Consumer& consumer)
 
                 const NextHopGroupKey& nhg = ctx.nhg;
 
-                if (ipv.size() == 1 && IpAddress(ipv[0]).isZero())
+                if (nhg.getSize() == 1 && nhg.hasIntfNextHop())
```

Comment:
Looks this changes is not compatible with previous. Previous check applies of Zero Address applies to both v4 and v6 but not new change applies onto to v4.

Reply:
@abdosi I did not write NextHopKey::isIntfNextHop()... it was already there.  I will update it to use .isZero() in its check unless you disagree.  That seems to be the more correct check.

Reply:
please update all places.

Reply:
@abdosi I am updating isIntfNextHop() function to use isZero(), not reverting the code here.  There is only one place to update... inside isIntfNextHop().